bluemira/equilibria/coils/_grouping.py

```diff
@@ -308,10 +309,21 @@ def from_group_vecs(cls, eqdsk: EQDSKInterface) -> CoilGroup:
                 "Please replace with an appropriate coilset."
             )
             return cls(*coils)
+
+        def _get_val(lst: npt.ArrayLike | None, idx: int, default=None):
+            if lst is None:
+                return None
+            try:
+                return lst[idx]
+            except IndexError:
+                return default
+
         for i in range(eqdsk.ncoil):
             dx = eqdsk.dxc[i]
             dz = eqdsk.dzc[i]
-            if abs(eqdsk.Ic[i]) < I_MIN:
+            cn = _get_val(eqdsk.coil_names, i)
+            ct = v if (v := _get_val(eqdsk.coil_types, i)) is None else CoilType(v)
```

Comment:
Just to clarify here, if `_get_val(eqdsk.coil_types, i)` results in None, ct will = None right?

Reply:
yep, avoiding passing None into coiltype